tools/tuf/releaser.sh

```diff
@@ -0,0 +1,305 @@
+#!/bin/bash
+
+#
+# For usage documentation, see the README.md.
+#
+
+set -e
+
+#
+# Input environment variables:
+#
+# AWS_PROFILE
+# TUF_DIRECTORY
+# COMPONENT
+# ACTION
+# VERSION
+# KEYS_SOURCE_DIRECTORY
+# TARGETS_PASSPHRASE_1PASSWORD_PATH
+# SNAPSHOT_PASSPHRASE_1PASSWORD_PATH
+# TIMESTAMP_PASSPHRASE_1PASSWORD_PATH
+# GITHUB_USERNAME
+# GITHUB_TOKEN_1PASSWORD_PATH
+# SKIP_PR_AND_TAG_PUSH
+#
+
+#
+# Dev environment variables:
+# PUSH_TO_REMOTE
+# GIT_REPOSITORY_DIRECTORY
+#
+
+clean_up () {
+    echo "Cleaning up directories..."
+
+    # Make sure (best effort) to remove the keys after we are done.
+    rm -rf "$KEYS_DIRECTORY"
+    rm -rf "$ARTIFACTS_DOWNLOAD_DIRECTORY"
+    rm -rf "$GO_TOOLS_DIRECTORY"
+    ARG=$?
+    exit $ARG
+} 
+
+setup () {
+    echo "Running setup..."
+
+    GO_TOOLS_DIRECTORY=$(mktemp -d)
+    ARTIFACTS_DOWNLOAD_DIRECTORY=$(mktemp -d)
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+    REPOSITORY_DIRECTORY=$TUF_DIRECTORY/repository
+    STAGED_DIRECTORY=$TUF_DIRECTORY/staged
+    KEYS_DIRECTORY=$TUF_DIRECTORY/keys
+    if [[ -z $GIT_REPOSITORY_DIRECTORY ]]; then
+        GIT_REPOSITORY_DIRECTORY=$( realpath "$SCRIPT_DIR/../.." )
+    fi
+
+    mkdir -p "$REPOSITORY_DIRECTORY"
+    mkdir -p "$STAGED_DIRECTORY"
+    cp -r "$KEYS_SOURCE_DIRECTORY" "$KEYS_DIRECTORY"
+
+    if ! aws sts get-caller-identity &> /dev/null; then
+        aws sso login
+        prompt "AWS SSO login was successful, press any key to continue..."
+    fi
+
+    GITHUB_TOKEN=$(op read "op://$GITHUB_TOKEN_1PASSWORD_PATH")
+
+    # These need to be exported for use by `fleetctl updates` commands.
+    FLEET_TARGETS_PASSPHRASE=$(op read "op://$TARGETS_PASSPHRASE_1PASSWORD_PATH")
+    export FLEET_TARGETS_PASSPHRASE
+    FLEET_SNAPSHOT_PASSPHRASE=$(op read "op://$SNAPSHOT_PASSPHRASE_1PASSWORD_PATH")
+    export FLEET_SNAPSHOT_PASSPHRASE
+    FLEET_TIMESTAMP_PASSPHRASE=$(op read "op://$TIMESTAMP_PASSPHRASE_1PASSWORD_PATH")
```

Comment:
Wouldn't it be more secure for `fleetctl updates` to interact with 1Password CLI?
Process env variables can be extracted with some tools.

Reply:
Good idea. Then you wouldn't have the passwords in plaintext in your console history.